### PR TITLE
tests: add snapshot_restore_pvpanic test

### PR DESCRIPTION
**Pull request by lisongqian**
Add test_snapshot_restore_pvpanic to integration tests since pvpanic already supports snapshot.

Signed-off-by: Songqian Li <sionli@tencent.com>

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -7719,6 +7719,173 @@ mod common_sequential {
 
         handle_child_output(r, &output);
     }
+
+    #[test]
+    fn test_snapshot_restore_pvpanic() {
+        _test_snapshot_restore_devices(true);
+    }
+    fn _test_snapshot_restore_devices(pvpanic: bool) {
```
Comment (liuw) [inline tests/integration.rs:7727]: Please leave a blank line before this function.

It feels a bit weird that you have a dedicated function but only use it once. Given the name of this function, do you plan to add more devices in the future?

Comment (lisongqian) [reply tests/integration.rs:7728]: Yes, I plan to add more devices in the future.

Comment (liuw) [reply tests/integration.rs:7728]: A new question: what's the rationale behind adding a new function than to extend the existing test helper? There is already one in the test suite for snapshotting.

Comment (lisongqian) [reply tests/integration.rs:7728]: I think they are in different scenes. `_test_snapshot_restore` ensures the **guest** can work normally after restoring. `_test_snapshot_restore_devices` ensures the **device** can work normally. And adding these codes will make the former bloated.

Comment (liuw) [reply tests/integration.rs:7729]: Okay. That's fair.

**Review by liuw — APPROVED** [review]

**Review by likebreath — APPROVED** [review]
@lisongqian Thank you for the contribution.

**Pull request merged**